### PR TITLE
GH-38847: [Documentation][C++] Explicitly note that compute is optional

### PR DESCRIPTION
**Pull request by bkietz**
### Rationale for this change

A user didn't know from reading just the compute documentation that compute is an optional feature. We can make that explicit

### What changes are included in this PR?

Added a cross-reference to the optional features section

### Are these changes tested?

No

### Are there any user-facing changes?

No

* GitHub Issue: #38847

### CODE REVIEW

File: docs/source/cpp/tutorials/compute_tutorial.rst [hunk 1]
```diff
@@ -41,6 +41,9 @@ Before continuing, make sure you have:
 
 1. An Arrow installation, which you can set up here: :doc:`/cpp/build_system`
 
+  - Be sure that your arrow installation
```
Comment (amoeba) [inline docs/source/cpp/tutorials/compute_tutorial.rst:44]:
```suggestion
  - Be sure that your Arrow installation
```

File: docs/source/cpp/tutorials/compute_tutorial.rst [hunk 2]
```diff
@@ -49,8 +52,8 @@ Setup
 Before running some computations, we need to fill in a couple gaps:
 
 1. We need to include necessary headers.
-
-2. ``A main()`` is needed to glue things together.
+/cpp/building.html#optional-components
```
Comment (amoeba) [inline docs/source/cpp/tutorials/compute_tutorial.rst:55]:
```suggestion
```

Was this added unintentionally?

File: docs/source/cpp/tutorials/compute_tutorial.rst [hunk 3]
```diff
@@ -41,6 +41,9 @@ Before continuing, make sure you have:
 
 1. An Arrow installation, which you can set up here: :doc:`/cpp/build_system`
```
Comment (amoeba) [inline docs/source/cpp/tutorials/compute_tutorial.rst:42]: Maybe just something like this instead?

```suggestion
1. An Arrow installation, which you can set up here: :doc:`/cpp/build_system`. If you're compiling Arrow yourself, be sure you compile with the compute module enabled (i.e., ``-DARROW_COMPUTE=ON``), see
:doc:`Optional Components. <cpp_build_optional_components>` for more information.
```

Then line 45 can be removed.

**Review by amoeba — CHANGES_REQUESTED** [review]
Feel free to leave my suggestion for bullet point (1).

File: docs/source/cpp/tutorials/compute_tutorial.rst (hunk 2 above)
Comment (bkietz) [reply docs/source/cpp/tutorials/compute_tutorial.rst:55]: Yes, thanks!

**Comment by mapleFU** [comment]
I remember a weird thing that even compute is not enabled, some module of compute (like basic cast) would be compiled. Should we mention this? Or just leave it here?

File: docs/source/cpp/tutorials/compute_tutorial.rst [hunk 4]
```diff
@@ -39,7 +39,9 @@ Pre-requisites
 
 Before continuing, make sure you have:
 
-1. An Arrow installation, which you can set up here: :doc:`/cpp/build_system`
+1. An Arrow installation, which you can set up here: :doc:`/cpp/build_system`.
+   If you're comiling Arrow yourself, be sure you compile with the compute module
```
[the remark this answers is not in the document]
Comment (bkietz) [reply docs/source/cpp/tutorials/compute_tutorial.rst:43]: argh

**Comment by bkietz** [comment]
> Should we mention this?

It might be worth documenting, but not in this tutorial since it uses kernels which are not included in that

**Comment by mapleFU** [comment]
+1 for me

**Review by amoeba — APPROVED** [review]

**Comment by amoeba** [comment]
LGTM. Thanks @bkietz.

**Pull request merged**